src/EnergyPlus/Autosizing/Base.cc

```diff
@@ -56,6 +57,45 @@
 
 namespace EnergyPlus {
 
+    bool oneTimeBaseSizerFlag = true;
+
+    void BaseSizer::clear_state() {
+        oneTimeBaseSizerFlag = true;
+    }
+
+    void BaseSizer::initializeWithinEP(EnergyPlusData &state, const std::string &_compType, const std::string &_compName, bool _printWarningFlag) {
+        this->printWarningFlag = _printWarningFlag;
+        this->compType = _compType;
+        this->compName = _compName;
+        this->sysSizingRunDone = DataSizing::SysSizingRunDone;
+        this->zoneSizingRunDone = DataSizing::ZoneSizingRunDone;
+        this->curSysNum = DataSizing::CurSysNum;
+        this->curOASysNum = DataSizing::CurOASysNum;
+        this->curZoneEqNum = DataSizing::CurZoneEqNum;
+        this->curDuctType = DataSizing::CurDuctType;
+        this->numPrimaryAirSys = DataHVACGlobals::NumPrimaryAirSys;
+        this->numSysSizInput = DataSizing::NumSysSizInput;
+        this->doSystemSizing = DataGlobals::DoSystemSizing;
+        this->numZoneSizingInput = DataSizing::NumZoneSizingInput;
+        this->doZoneSizing = DataGlobals::DoZoneSizing;
+        this->curTermUnitSizingNum = DataSizing::CurTermUnitSizingNum;
+        this->termUnitSingDuct = DataSizing::TermUnitSingDuct;
+        this->termUnitPIU = DataSizing::TermUnitPIU;
+        this->termUnitIU = DataSizing::TermUnitIU;
+        this->zoneEqFanCoil = DataSizing::ZoneEqFanCoil;
+        this->otherEqType = !(this->termUnitSingDuct || this->termUnitPIU || this->termUnitIU || this->zoneEqFanCoil);
+        this->zoneSizingInput = DataSizing::ZoneSizingInput;
+        this->unitarySysEqSizing = DataSizing::UnitarySysEqSizing;
+        this->oaSysEqSizing = DataSizing::OASysEqSizing;
+        this->outsideAirSys = DataAirLoop::OutsideAirSys;
+        this->termUnitSizing = DataSizing::TermUnitSizing;
+        this->finalZoneSizing = DataSizing::FinalZoneSizing;
+        this->zoneEqSizing = DataSizing::ZoneEqSizing;
+        this->sysSizingInputData = DataSizing::SysSizInput;
+        this->finalSysSizing = DataSizing::FinalSysSizing;
+        this->airloopDOAS = AirLoopHVACDOAS::airloopDOAS;
+    }
```

Comment:
@rraustad I moved all this to a base class initialize function so that each sizer class wouldn't need to replicate it.